tests/functional/test-offer-fail

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+tmpfile=$(mktemp ./leaseXXXXXX)
+errfile=$(mktemp ./errXXXXXX)
+start=$(date +%Y-%m-%d)
+end=$(date -d "+5 days" +%Y-%m-%d)
+lessee_uuid=39b16f5117c24072844d55f8d20cea5d
+
+./setup $tmpfile $start $end $lessee_uuid
+
+. $tmpfile
+
+test_offer_uuid=$uuid
+trap "rm -f $tmpfile $errfile; ./cleanup $test_offer_uuid" EXIT  # Run cleanup if the script exits
+
+echo "OFFER CREATE TIME CONFLICT TEST"
+
+. "$ESI_TEST_CREDENTIALS/test1-openrc"
+openstack esi offer create \
+  --start-time $start \
+  --end-time $end \
+  --lessee-id $lessee_uuid \
+  --resource-type dummy_node 1001 \
+  -f shell > $tmpfile 2> $errfile
+ec=$?
+expected_error='Time conflict for dummy_node 1001.'
+
+if ! cat $errfile | grep -q "$expected_error"; then
+  if [[ $ec -eq 0 ]]; then
+    echo "ERROR: succeeded creating offer despite time conflict" >&2
+    . $tmpfile
+    echo "Deleting time conflict test offer..."
+    ./cleanup $uuid
+  else
+    echo "ERROR: unexpected error during time conflict test" >&2
+    cat $errfile >&2
+  fi
+  exit 1
+fi
+
+echo "OFFER CREATE TIME CONFLICT TEST SUCCEEDED"
+
+# Setting up a claim is needed for the claim time conflict test
+
+echo "OFFER CLAIM TEST"
```

Comment:
You can other claim_offer_fail cases in this script such as lessee outside test1 can't view or claim the offer